From 1f7ec532b80c4c420bdd7fb11afc53cda116925b Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Mon, 10 Feb 2014 10:06:36 +0800
Subject: [PATCH 58/58] ENGR00295554 ethernet:add proxy function

Add ethernet proxy function in advance setting.
test method:
    --Not set proxy, open google.com in internal networking,
      it will auto redirect to google.com.hk.
    --Set proxy, open google.com in internal networking,
      it will open google.com instead of google.com.hk.
    --Also support domain name in proxy ip. But need to set DNS
      server to a proper one which can resolve it.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetAdvDialog.java | 13 ++-----------
 Ethernet/com/fsl/ethernet/EthernetManager.java   |  4 ++--
 2 files changed, 4 insertions(+), 13 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
index 3da7895..e6473da 100644
--- a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
@@ -76,7 +76,6 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
     }
 
     public void handle_saveconf() {
-        //mLinkProperties.clear();
         EthernetDevInfo info = new EthernetDevInfo();
 
         if ((mProxyIp.getText().toString().equals(""))||(mProxyPort.getText().toString().equals("")))
@@ -89,17 +88,9 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
             info.setProxyAddr(mProxyIp.getText().toString());
             info.setProxyPort(mProxyPort.getText().toString());
             info.setProxyExclusionList(mProxyExclusionList.getText().toString());
-        } mEthEnabler.getManager().updateDevInfo(info);
-
-        String proxyAddress = mProxyIp.getText().toString();
-        String proxyPort = mProxyPort.getText().toString();
-        String exclusionList = mProxyExclusionList.getText().toString();
-        int port = 0;
-        try {
-            port = Integer.parseInt(proxyPort);
-        } catch(NumberFormatException e){
         }
-        Proxy.setHttpProxySystemProperty(proxyAddress,proxyPort,exclusionList,null);
+        mEthEnabler.getManager().updateDevInfo(info);
+        mEthEnabler.getManager().setProxy( );
     }
 
     public void onClick(DialogInterface dialog, int which) {
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index 7881e4b..0cbc4d9 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -376,8 +376,8 @@ public class EthernetManager {
         }
         ProxyProperties proxyProperties =
             new ProxyProperties(getSharedPreProxyAddress(), port, exclusionList);
-        lp.setHttpProxy(proxyProperties);
-        Log.i(TAG,"=============getHttpProxy==============" + lp.getHttpProxy());
+        mConnMgr.setGlobalProxy(proxyProperties);
+        Log.i(TAG,"=============getHttpProxy==============" + proxyProperties);
     }
 
 }
-- 
1.8.0

